### PR TITLE
Use selector instead of signature in story runner

### PR DESCRIPTION
**Pull request by eccles**
Problem:
The use of an asset or location signature was considered a
inferior design. Use a selector instead where one can specify
which attributes signify a unique asset/location.

Solution:
Changed create_of_not_exists() of both assets and locations to
use a selector. Modified door_entry story.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: functests/test_resources/door_entry_story.yaml [hunk 1]
```diff
@@ -8,25 +8,29 @@
...
-    signature:
-      attributes:
-        arc_display_name: Jitsuin Paris Front Door
+    selector:
+      - attributes.arc_display_name
     behaviours:
       - RecordEvidence
       - Attachments
     attributes:
+      arc_display_name: Jitsuin Paris Front Door 2022-03-04
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:24]: Probably didn't mean to commit this funky name. There are a bunch of these I won't point them all out.

File: functests/test_resources/door_entry_story.yaml [hunk 2]
```diff
@@ -71,21 +76,22 @@ steps:
...
       arc_display_type: door
       arc_firmware_version: "1.0"
       arc_serial_number: das-x4-02
       arc_description: Electronic door entry system controlling the main staff entrance to Paris Courts of Justice
       wavestone_asset_id: courts.paris.wavestonedas
     location:
-      signature:
-        display_name: Paris Courts of Justice
+      selector:
+        - display_name
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:93]: arc_

File: functests/test_resources/door_entry_story.yaml [hunk 3]
```diff
@@ -71,21 +76,22 @@ steps:
...
+      display_name: Paris Courts of Justice 2022-03-04
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:94]: arc_

File: functests/test_resources/door_entry_story.yaml [hunk 4]
```diff
@@ -100,21 +106,22 @@ steps:
   - step:
       action: ASSETS_CREATE_IF_NOT_EXISTS
       description: Create a door in Paris at Bastille
-    signature:
-      attributes:
-        arc_display_name: Bastille Paris Front Door
+    selector:
+      - attributes.arc_display_name
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:110]: Doesn't need attributes. (?)

File: functests/test_resources/door_entry_story.yaml [hunk 5]
```diff
@@ -100,21 +106,22 @@ steps:
...
       arc_display_type: door
       arc_firmware_version: "1.0"
       arc_serial_number: das-x4-03
       arc_description: Electronic door entry system controlling the main staff entrance to Bastille
       wavestone_asset_id: bastille.paris.wavestonedas
     location:
-      signature:
-        display_name: Bastille
+      selector:
+        - display_name
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:123]: arc_

File: functests/test_resources/door_entry_story.yaml [hunk 6]
```diff
@@ -100,21 +106,22 @@ steps:
...
+      display_name: Bastille 2022-03-04
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:124]: arc_

File: functests/test_resources/door_entry_story.yaml [hunk 7]
```diff
@@ -129,21 +136,22 @@ steps:
   - step:
       action: ASSETS_CREATE_IF_NOT_EXISTS
       description: Create a front door in Paris at Gare du Nord Apartments
-    signature:
-      attributes:
-        arc_display_name: Gare du Nord Apartments Paris Front Door
+    selector:
+      - attributes.arc_display_name
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:140]: 'attributes.' ?

File: functests/test_resources/door_entry_story.yaml [hunk 8]
```diff
@@ -129,21 +136,22 @@ steps:
...
       arc_display_type: door
       arc_firmware_version: "1.0"
       arc_serial_number: das-x4-04
       arc_description: Electronic door entry system controlling the front residential entrance to Apartements du Gare du Nord
       wavestone_asset_id: front.gdn.paris.wavestonedas
     location:
-      signature:
-        display_name: Apartements du Gare du Nord
+      selector:
+        - display_name
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:153]: arc_

File: functests/test_resources/door_entry_story.yaml [hunk 9]
```diff
@@ -129,21 +136,22 @@ steps:
...
+      display_name: Apartements du Gare du Nord 2022-03-04
```
Comment (JAG-UK) [inline functests/test_resources/door_entry_story.yaml:154]: arc_ again. I'm going to stop pointing these out but there are more

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: functests/test_resources/door_entry_story.yaml (hunk 9 above)
Comment (eccles) [reply functests/test_resources/door_entry_story.yaml:158]: they are all the display name of 'location' which does not have arc_ prefix.....

File: functests/test_resources/door_entry_story.yaml (hunk 1 above)
Comment (eccles) [reply functests/test_resources/door_entry_story.yaml:25]: these have been replaced with getting a namespace from environment in the estate info PR

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**